reyna/src/com/b2msolutions/reyna/messageProvider/BatchProvider.java

```diff
@@ -34,6 +36,11 @@ public BatchProvider(Context context, Repository repository) {
     public Message getNext() throws URISyntaxException {
         Logger.v(BatchProvider.TAG, "getNext");
 
+        if (exitAsLastBatchWasNotFull) {
+            Logger.v(TAG, "getNext exit as there is no enough messages to form a batch");
```

Comment:
Do we have to set it back to false, or are we just having a new BatchProvider the next time we trigger sending?